src/EllCrv/NewtonLift.jl

```diff
@@ -0,0 +1,295 @@
+    f = (x^3 + a2c*x^2 + a4c*x + a6c)
+    pt = [i[1] for i in factor(f) if i[2]>1]
+    @assert length(pt)==1
+
+    pt = pt[1]
+    x0 = -constant_coefficient(pt)
+    y0 = F(0)
+    push!(sing, [x0,y0,t0])
+  end
+  return sing
+end
+
+@doc r"""
+    newton_lift(E::EllipticCurve, P::EllipticCurvePoint, reduction_map; max_iterations=16) -> Bool, EllipticCurvePoint
+
+Return a point `Q` of `E` which reduces by `reduction_map` to `P` and whether it exists.
+
+Let ``F`` be a number field and ``E`` an elliptic curve over ``F(t)`` which is integral over ``F[t]``.
+Let ``O_F`` be the maximal order of ``F`` and ``p`` be a prime ideal of ``O_F`` with residue field ``k``.
+Suppose that ``E`` is integral over ``(O_F)_p`` and that it has good reduction at ``p``.
+
+The algorithm lifts `P` by a multivariate Newton iteration to a point in the completion of ``F`` at the prime ``p``
+(up to some finite precision). The corresponding point of small height in `F` is computed and the result verified.
+
+Throws an error if `max_iterations` is reached without recognizing the point in `F`.
+Use `set_verbosity_level(:NewtonLift, 2)` for output during the computation.
+
+Input:
+- `E` -- the elliptic curve ``E/F(t)``
+- `p` -- a point of ``E/k(t)``
+- `reduction map` -- the canonical homomorphism $O_F \rightarrow k$
+- max_iterations -- the maximal number of Newton iteration steps
+"""
+function newton_lift(E::EllipticCurve{<:Generic.FracFieldElem{<:PolyRingElem}}, P::EllipticCurvePoint{<:Generic.FracFieldElem{<:PolyRingElem}}, reduction_map::Map; max_iterations::Int=16)
+  # O_F -> O_F/m =: k
+  # K the completion of F at m
+  sing = _singular_points_weierstrass_model(E)
+  @req codomain(reduction_map)===base_ring(base_ring(base_field(parent(P)))) "codomain of the reduction map incompatible with the point to be lifted"
+  @req number_field(domain(reduction_map))===base_ring(base_ring(base_field(E))) "codomain of the reduction map incompatible with the point to be lifted"
+  @req max_iterations <= 62 " the desired precision needs to fit into an Int64"
+  prime = reduction_map.P
+  F = number_field(domain(reduction_map))
+  F_to_k = extend(reduction_map, F)
+  K, F_to_K = completion(F, prime, 2^max_iterations)
+  OK = valuation_ring(K)
+  Ft = base_ring(base_field(E))
+  OKt,_ = polynomial_ring(OK,'t'; cached=false)
+  F_to_OK = map_from_func(x-> OK(F_to_K(x)), F, OK)
+  Ft_to_OKt = map_from_func(x->map_coefficients(F_to_OK, x; parent=OKt), Ft, OKt)
+
+  return  _newton_lift(P, E, F_to_K, Ft_to_OKt, F_to_k, max_iterations, sing, prime)
+end
+
+
+function _can_reconstruct_with_reconstruction(E::EllipticCurve, padic_point, F_to_K, prec)
+  @vprint :NewtonIteration 2 "trying rational reconstruction\n"
+  K = codomain(F_to_K)
+  # prec0 = precision(K)
+  #setprecision!(F_to_K, prec) # leads to difficult bugs
+
+  xnF, ynF, dnF = [map_coefficients(x->preimage(F_to_K,K(x); small_lift=true), a) for a in padic_point]
+
+  #setprecision!(F_to_K, prec0)
```

Comment:
@joschmitt  thanks to your work, I could get this pr running. 

In this bit the `setprecision!` method fails and I am out of my depth fixing it. 
would you have a look? @fieker 

(merely removing the assertion does not help.)